src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemEventArgs.cs

```diff
@@ -17,14 +17,18 @@ public class FileSystemEventArgs : EventArgs
         /// </devdoc>
         public FileSystemEventArgs(WatcherChangeTypes changeType, string directory, string? name)
         {
+            ArgumentNullException.ThrowIfNull(directory);
```

Comment:
This is just lifting out the check that was previously being performed by GetFullPath?

Reply:
> This is just lifting out the check that was previously being performed by GetFullPath?

Correct, since that would no longer get called with a null.